index.js

```diff
@@ -112,3 +112,14 @@ MochaJUnitReporter.prototype.writeXmlToDisk = function(xml){
   fs.writeFileSync(filePath, xml, 'utf-8');
   console.log('test results written to', filePath);
 };
+
+
+function ISODateString(d) {
```

Comment:
You're right, I assumed (incorrectly) that the length of the milliseconds part was not always the same. To be sure I did some tests and there are always 3 digits used for the milliseconds. I now implemented your suggestion.